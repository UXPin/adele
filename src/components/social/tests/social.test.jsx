import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import Social from '../social';

it('renders three Social buttons correctly', () => {
  const wrapper = mount(<Social networks={['facebook', 'twitter', 'linkedin']} url={'http://#'}/>);
  expect(wrapper).toMatchSnapshot();
});

it('renders three Social buttons correctly', () => {
  const wrapper = mount(<Social networks={['facebook', 'twitter']} url={'http://#'}/>);
  expect(wrapper).toMatchSnapshot();
});
