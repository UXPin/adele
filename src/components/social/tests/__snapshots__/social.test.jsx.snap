// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders three Social buttons correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 6px;
}

.c0 [title~='facebook'] span {
  padding: 0 0 1px 2px;
}

.c0 [title~='twitter'] span {
  padding: 0 0 0 2px;
}

.c0 [title~='linkedin'] span {
  padding: 0 0 0 2px;
}

.c0 [title~='github'] span {
  padding: 0 0 1px 2px;
}

.c0 [title~='github'] span svg {
  width: 20px;
  margin-left: -7px;
}

.c1 {
  display: inline-block;
  margin-left: 6px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  background: none;
  border: 1px solid #006cff;
  border-radius: 100%;
  -webkit-transition: background-color 0.3s ease-in;
  transition: background-color 0.3s ease-in;
  cursor: pointer;
  outline: none;
}

.c2 span {
  position: relative;
  display: block;
  width: 14px;
  height: 14px;
}

.c2 svg {
  display: block;
  -webkit-transition: fill 0.3s ease-in;
  transition: fill 0.3s ease-in;
  width: 10px;
  height: 14px;
}

.c2:hover,
.c2:focus {
  background-color: #006cff;
}

.c2:hover svg,
.c2:focus svg {
  fill: white;
}

.c3:focus {
  outline: 0;
}

.c3 > svg {
  width: 14px;
  height: 100%;
  fill: #006cff;
  cursor: pointer;
}

.c3 > svg:hover {
  fill: #006cff;
}

.c4:focus {
  outline: 0;
}

.c4 > svg {
  width: 13px;
  height: 100%;
  fill: #006cff;
  cursor: pointer;
}

.c4 > svg:hover {
  fill: #006cff;
}

<Social
  networks={
    Array [
      "facebook",
      "twitter",
      "linkedin",
    ]
  }
  tab={0}
  url="http://#"
>
  <styled.ul>
    <ul
      className="c0"
    >
      <styled.li
        title="github"
      >
        <li
          className="c1"
          title="github"
        >
          <styled.span
            tabIndex={0}
          >
            <span
              className="c2"
              tabIndex={0}
            >
              <a
                href="https://github.com/uxpin/adele"
                rel="noopener noreferrer"
                target="_blank"
              >
                <Icon
                  active={true}
                  color="#006cff"
                  i="test-file-stub"
                  in="blue"
                  rotate={0}
                  size="m"
                >
                  <Styled(SVGInline)
                    color="#006cff"
                    data-active={true}
                    in="blue"
                    rotate={0}
                    size="m"
                    svg="test-file-stub"
                  >
                    <SVGInline
                      className="c3"
                      classSuffix="-svg"
                      cleanup={Array []}
                      cleanupExceptions={Array []}
                      color="#006cff"
                      component="span"
                      data-active={true}
                      in="blue"
                      rotate={0}
                      size="m"
                      svg="test-file-stub"
                    >
                      <span
                        className="SVGInline c3"
                        color="#006cff"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "test-file-stub",
                          }
                        }
                        data-active={true}
                        in="blue"
                        rotate={0}
                        size="m"
                      />
                    </SVGInline>
                  </Styled(SVGInline)>
                </Icon>
              </a>
            </span>
          </styled.span>
        </li>
      </styled.li>
      <styled.li
        key="facebook"
        title="facebook social share"
      >
        <li
          className="c1"
          title="facebook social share"
        >
          <styled.span
            onClick={[Function]}
            tabIndex={0}
          >
            <span
              className="c2"
              onClick={[Function]}
              tabIndex={0}
            >
              <Icon
                active={true}
                color="#006cff"
                i="test-file-stub"
                in="blue"
                rotate={0}
                size="s"
              >
                <Styled(SVGInline)
                  color="#006cff"
                  data-active={true}
                  in="blue"
                  rotate={0}
                  size="s"
                  svg="test-file-stub"
                >
                  <SVGInline
                    className="c4"
                    classSuffix="-svg"
                    cleanup={Array []}
                    cleanupExceptions={Array []}
                    color="#006cff"
                    component="span"
                    data-active={true}
                    in="blue"
                    rotate={0}
                    size="s"
                    svg="test-file-stub"
                  >
                    <span
                      className="SVGInline c4"
                      color="#006cff"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "test-file-stub",
                        }
                      }
                      data-active={true}
                      in="blue"
                      rotate={0}
                      size="s"
                    />
                  </SVGInline>
                </Styled(SVGInline)>
              </Icon>
            </span>
          </styled.span>
        </li>
      </styled.li>
      <styled.li
        key="twitter"
        title="twitter social share"
      >
        <li
          className="c1"
          title="twitter social share"
        >
          <styled.span
            onClick={[Function]}
            tabIndex={0}
          >
            <span
              className="c2"
              onClick={[Function]}
              tabIndex={0}
            >
              <Icon
                active={true}
                color="#006cff"
                i="test-file-stub"
                in="blue"
                rotate={0}
                size="s"
              >
                <Styled(SVGInline)
                  color="#006cff"
                  data-active={true}
                  in="blue"
                  rotate={0}
                  size="s"
                  svg="test-file-stub"
                >
                  <SVGInline
                    className="c4"
                    classSuffix="-svg"
                    cleanup={Array []}
                    cleanupExceptions={Array []}
                    color="#006cff"
                    component="span"
                    data-active={true}
                    in="blue"
                    rotate={0}
                    size="s"
                    svg="test-file-stub"
                  >
                    <span
                      className="SVGInline c4"
                      color="#006cff"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "test-file-stub",
                        }
                      }
                      data-active={true}
                      in="blue"
                      rotate={0}
                      size="s"
                    />
                  </SVGInline>
                </Styled(SVGInline)>
              </Icon>
            </span>
          </styled.span>
        </li>
      </styled.li>
      <styled.li
        key="linkedin"
        title="linkedin social share"
      >
        <li
          className="c1"
          title="linkedin social share"
        >
          <styled.span
            onClick={[Function]}
            tabIndex={0}
          >
            <span
              className="c2"
              onClick={[Function]}
              tabIndex={0}
            >
              <Icon
                active={true}
                color="#006cff"
                i="test-file-stub"
                in="blue"
                rotate={0}
                size="s"
              >
                <Styled(SVGInline)
                  color="#006cff"
                  data-active={true}
                  in="blue"
                  rotate={0}
                  size="s"
                  svg="test-file-stub"
                >
                  <SVGInline
                    className="c4"
                    classSuffix="-svg"
                    cleanup={Array []}
                    cleanupExceptions={Array []}
                    color="#006cff"
                    component="span"
                    data-active={true}
                    in="blue"
                    rotate={0}
                    size="s"
                    svg="test-file-stub"
                  >
                    <span
                      className="SVGInline c4"
                      color="#006cff"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "test-file-stub",
                        }
                      }
                      data-active={true}
                      in="blue"
                      rotate={0}
                      size="s"
                    />
                  </SVGInline>
                </Styled(SVGInline)>
              </Icon>
            </span>
          </styled.span>
        </li>
      </styled.li>
    </ul>
  </styled.ul>
</Social>
`;

exports[`renders three Social buttons correctly 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 6px;
}

.c0 [title~='facebook'] span {
  padding: 0 0 1px 2px;
}

.c0 [title~='twitter'] span {
  padding: 0 0 0 2px;
}

.c0 [title~='linkedin'] span {
  padding: 0 0 0 2px;
}

.c0 [title~='github'] span {
  padding: 0 0 1px 2px;
}

.c0 [title~='github'] span svg {
  width: 20px;
  margin-left: -7px;
}

.c1 {
  display: inline-block;
  margin-left: 6px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  background: none;
  border: 1px solid #006cff;
  border-radius: 100%;
  -webkit-transition: background-color 0.3s ease-in;
  transition: background-color 0.3s ease-in;
  cursor: pointer;
  outline: none;
}

.c2 span {
  position: relative;
  display: block;
  width: 14px;
  height: 14px;
}

.c2 svg {
  display: block;
  -webkit-transition: fill 0.3s ease-in;
  transition: fill 0.3s ease-in;
  width: 10px;
  height: 14px;
}

.c2:hover,
.c2:focus {
  background-color: #006cff;
}

.c2:hover svg,
.c2:focus svg {
  fill: white;
}

.c3:focus {
  outline: 0;
}

.c3 > svg {
  width: 14px;
  height: 100%;
  fill: #006cff;
  cursor: pointer;
}

.c3 > svg:hover {
  fill: #006cff;
}

.c4:focus {
  outline: 0;
}

.c4 > svg {
  width: 13px;
  height: 100%;
  fill: #006cff;
  cursor: pointer;
}

.c4 > svg:hover {
  fill: #006cff;
}

<Social
  networks={
    Array [
      "facebook",
      "twitter",
    ]
  }
  tab={0}
  url="http://#"
>
  <styled.ul>
    <ul
      className="c0"
    >
      <styled.li
        title="github"
      >
        <li
          className="c1"
          title="github"
        >
          <styled.span
            tabIndex={0}
          >
            <span
              className="c2"
              tabIndex={0}
            >
              <a
                href="https://github.com/uxpin/adele"
                rel="noopener noreferrer"
                target="_blank"
              >
                <Icon
                  active={true}
                  color="#006cff"
                  i="test-file-stub"
                  in="blue"
                  rotate={0}
                  size="m"
                >
                  <Styled(SVGInline)
                    color="#006cff"
                    data-active={true}
                    in="blue"
                    rotate={0}
                    size="m"
                    svg="test-file-stub"
                  >
                    <SVGInline
                      className="c3"
                      classSuffix="-svg"
                      cleanup={Array []}
                      cleanupExceptions={Array []}
                      color="#006cff"
                      component="span"
                      data-active={true}
                      in="blue"
                      rotate={0}
                      size="m"
                      svg="test-file-stub"
                    >
                      <span
                        className="SVGInline c3"
                        color="#006cff"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "test-file-stub",
                          }
                        }
                        data-active={true}
                        in="blue"
                        rotate={0}
                        size="m"
                      />
                    </SVGInline>
                  </Styled(SVGInline)>
                </Icon>
              </a>
            </span>
          </styled.span>
        </li>
      </styled.li>
      <styled.li
        key="facebook"
        title="facebook social share"
      >
        <li
          className="c1"
          title="facebook social share"
        >
          <styled.span
            onClick={[Function]}
            tabIndex={0}
          >
            <span
              className="c2"
              onClick={[Function]}
              tabIndex={0}
            >
              <Icon
                active={true}
                color="#006cff"
                i="test-file-stub"
                in="blue"
                rotate={0}
                size="s"
              >
                <Styled(SVGInline)
                  color="#006cff"
                  data-active={true}
                  in="blue"
                  rotate={0}
                  size="s"
                  svg="test-file-stub"
                >
                  <SVGInline
                    className="c4"
                    classSuffix="-svg"
                    cleanup={Array []}
                    cleanupExceptions={Array []}
                    color="#006cff"
                    component="span"
                    data-active={true}
                    in="blue"
                    rotate={0}
                    size="s"
                    svg="test-file-stub"
                  >
                    <span
                      className="SVGInline c4"
                      color="#006cff"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "test-file-stub",
                        }
                      }
                      data-active={true}
                      in="blue"
                      rotate={0}
                      size="s"
                    />
                  </SVGInline>
                </Styled(SVGInline)>
              </Icon>
            </span>
          </styled.span>
        </li>
      </styled.li>
      <styled.li
        key="twitter"
        title="twitter social share"
      >
        <li
          className="c1"
          title="twitter social share"
        >
          <styled.span
            onClick={[Function]}
            tabIndex={0}
          >
            <span
              className="c2"
              onClick={[Function]}
              tabIndex={0}
            >
              <Icon
                active={true}
                color="#006cff"
                i="test-file-stub"
                in="blue"
                rotate={0}
                size="s"
              >
                <Styled(SVGInline)
                  color="#006cff"
                  data-active={true}
                  in="blue"
                  rotate={0}
                  size="s"
                  svg="test-file-stub"
                >
                  <SVGInline
                    className="c4"
                    classSuffix="-svg"
                    cleanup={Array []}
                    cleanupExceptions={Array []}
                    color="#006cff"
                    component="span"
                    data-active={true}
                    in="blue"
                    rotate={0}
                    size="s"
                    svg="test-file-stub"
                  >
                    <span
                      className="SVGInline c4"
                      color="#006cff"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "test-file-stub",
                        }
                      }
                      data-active={true}
                      in="blue"
                      rotate={0}
                      size="s"
                    />
                  </SVGInline>
                </Styled(SVGInline)>
              </Icon>
            </span>
          </styled.span>
        </li>
      </styled.li>
    </ul>
  </styled.ul>
</Social>
`;
